utils.py

```diff
@@ -471,3 +471,19 @@ def get_asset_dir_prefix():
             ASSET_DIR_PREFIX = '/build/%s' % cache_slug
 
     return ASSET_DIR_PREFIX
+
+TEMPLATE_DIR_PREFIX = None
+def get_template_dir_prefix():
+    """Returns prefix for template directory depending whether dev or prod.
+    It is used as a prefix in urls for script files.
+    """
+    global TEMPLATE_DIR_PREFIX # pylint: disable=global-statement
+    if not TEMPLATE_DIR_PREFIX:
+        TEMPLATE_DIR_PREFIX = '/templates/dev/head'
```

Comment:
I seem to remember there was an issue previously with get_asset_dir_prefix() that meant that ASSET_DIR_PREFIX was set to the empty string before feconf.IS_MINIFIED became True. So we changed the check from "X is not None" to "if not X".

Here, is there a danger that TEMPLATE_DIR_PREFIX will be set to this constant value prematurely?

If so, here's a suggestion: have CACHE_SLUG be the global variable, have a get_cache_slug() method, and let get_asset_dir_prefix() etc. calculate the asset/template directory prefixes on the fly by referencing get_cache_slug(). That removes code duplication and also avoids an ugly `if not TEMPLATE_DIR_PREFIX or TEMPLATE_DIR_PREFIX = '/templates/dev/head'` check.